IndomitableIbises/Bangazon-Sprint-3
# Sell product ea

## Link to Ticket
Closes # [Link](https://github.com/IndomitableIbises/Bangazon-Sprint-3/issues/21)

## Description of Proposed Changes

This pull request adds:
  1. A detail view from the product list
  2. A detail view from the sell a product screen
  3.  Hides the "sell a product" affordance for unauthenticated users

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout sellProduct-ea
```
If you have faker:
1.  stop running server
2.  close db browser
3. in terminal type "django_data website faker_factory" (for MAC) or "./django_data website faker_factory" (for pc users)

if you don't have faker:
1.  heaven help you

then:
1.  python manage.py runserver
2. if you aren't logged in you should NOT see the sell a product affordance
3.  register a new user.  
4.  click on the sell a product affordance and add a product to sell
5.  you should see a detailed view of the product now added 
6.  from that screen click the product list affordance
7.  click on a product and you will see a detailed view of the product



## Impacted Areas in Application

List general components of the application that this PR will affect:

* sell a product detail, product list

## Mentions @username

Tag users that need to review this code
@rafcevallos , @paulzimmclay 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
